src/lib/provable/int.ts

```diff
@@ -532,6 +534,68 @@ class UInt64 extends CircuitValue {
   static fromBytesBE(bytes: UInt8[]): UInt64 {
     return UInt64.fromBytes([...bytes].reverse());
   }
+
+  /**
+   * Returns an array of {@link Bool} elements representing [little endian binary representation](https://en.wikipedia.org/wiki/Endianness) of this {@link UInt64} element.
+   *
+   * If you use the optional `length` argument, proves that the UInt64 element fits in `length` bits.
+   * The `length` has to be between 0 and 64 and the method throws if it isn't.
+   *
+   * **Warning**: The cost of this operation in a zk proof depends on the `length` you specify,
+   * which by default is 64 bits. Prefer to pass a smaller `length` if possible.
+   *
+   * @param length - the number of bits to fit the element. If the element does not fit in `length` bits, the functions throws an error.
+   *
+   * @return An array of {@link Bool} element representing little endian binary representation of this {@link UInt64}.
+   */
+  toBits(length: number = 64) {
+    checkBitLength('UInt64.toBits()', length, 64);
+    if (this.isConstant()) {
+      let bits = BinableFp.toBits(this.toBigInt());
+      if (bits.slice(length).some((bit) => bit))
+        throw Error(`UInt64.toBits(): ${this} does not fit in ${length} bits`);
+      return bits.slice(0, length).map((b) => new Bool(b));
+    }
+    let bits = Provable.witness(Provable.Array(Bool, length), () => {
+      let f = this.toBigInt();
+      return Array.from({ length }, (_, k) => new Bool(!!((f >> BigInt(k)) & 0x1n)));
+    });
+    UInt64.fromBits(bits).assertEquals(this);
+    return bits;
```

Comment:
I don't get why you don't use the `field.toBits()` method here?

```suggestion
    checkBitLength('UInt64.toBits()', length, 64);
    return this.value.toBits(length);
```

Reply:
In this case, since we can't customize the part
```ts
if (this.isConstant()) {
  let bits = BinableFp.toBits(this.toBigInt());
  if (bits.slice(length).some((bit) => bit))
    throw Error(`Field.toBits(): ${this} does not fit in ${length} bits`); // this error message
  return bits.slice(0, length).map((b) => new Bool(b));
}
```
in the `toBits` method for the `UInt32` and `UInt64` classes, error messages will only be shown for the `Field` type, which might be confusing for developers. wdyt?

Reply:
yep that's a good point! but you could at least use it in the non-constant branch